.github/workflows/infra-ci.yml

```diff
@@ -0,0 +1,68 @@
+name: Infrastructure CI
```

Comment:
small nit — should we name this something more explicit like `infra-lint.yml` or `infra-checks.yml`? I think that's one of my regrets from PFML, since it's difficult to rename the workflow files afterwards 😓 and "ci" can mean different things to different folks.